cmd/xl-storage-format-utils.go

```diff
@@ -30,15 +30,12 @@ func getFileInfoVersions(xlMetaBuf []byte, volume, path string, allParts bool) (
 	}
 	n := 0
 	for _, fi := range fivs.Versions {
-		// Filter our tier object delete marker
-		if !fi.TierFreeVersion() {
-			fivs.Versions[n] = fi
-			n++
-		} else {
+		if fi.TierFreeVersion() {
 			fivs.FreeVersions = append(fivs.FreeVersions, fi)
+		} else {
```

Comment:
`applyVersionActions` (from scanner) expects `getFileInfoVersions` to return `fivs FileInfoVersions` such that, `fivs.Versions` contains only non-free versions and `fivs.FreeVersions` contains only free versions. 
This is used when applying `NewerNoncurrentVersions` lifecycle element.

To address this, we can extend `getFileInfoVersions` to take an argument, `inclFreeVersions bool`, based on which we can populate `fivs.Versions` to include free versions (or not).
```